shared/src/ledger/eth_bridge/vp/mod.rs

```diff
@@ -27,6 +28,56 @@ where
     pub ctx: Ctx<'ctx, DB, H, CA>,
 }
 
+impl<'ctx, DB, H, CA> EthBridge<'ctx, DB, H, CA>
+where
+    DB: 'static + ledger_storage::DB + for<'iter> ledger_storage::DBIter<'iter>,
+    H: 'static + StorageHasher,
+    CA: 'static + WasmCacheAccess,
+{
+    /// If the bridge's escrow key was changed, we check
+    /// that the balance increased and that the bridge pool
+    /// VP has been triggered. The bridge pool VP will carry
+    /// out the rest of the checks.
+    fn check_escrow(&self, verifiers: &BTreeSet<Address>) -> bool {
```

Comment:
Would suggest return `Result<bool>` here as well. Specifically where we're currently doing `.expect`, we should propagate an `Err`, also where we can't retrieve the `escrow_key` value.

Reply:
The error in this case means we can't read our DB. Is that a recoverable error? I think it's better to panic in such a case.

Reply:
It's not recoverable but I don't think panicking here is the best way to stop the process, we could propagate the error upwards, and then take a decision somewhere to shut down gracefully based on that unrecoverable error. We don't have functionality like that right now in the ledger though so have to `.unwrap()` at some point in our call path. I guess it is a bigger issue than this PR.